Bump extensions hosted on GitHub

niklaskeerl opened the pull request:
This commit bumps all extensions hosted on GitHub to the newest version.

jankeromnes commented on extensions.json:
```diff
@@ -900,8 +900,8 @@
     {
       "id": "platformio.platformio-ide",
       "repository": "https://github.com/platformio/platformio-vscode-ide",
```
I notice that this extension made a pre-packaged `.vsix` version available.

In these cases, we prefer re-publishing official `.vsix` files rather than trying to re-package it ourselves (bears the risk of doing something wrong and uploading a broken extension to OpenVSX, e.g. if we forget an important packaging step).

```suggestion
      "download": "https://github.com/platformio/platformio-vscode-ide/releases/download/v2.1.0/platformio-ide-2.1.0.vsix",
```

jankeromnes commented on extensions.json:
```diff
@@ -900,8 +900,8 @@
       "repository": "https://github.com/platformio/platformio-vscode-ide",
-      "version": "1.10.0",
-      "checkout": "v1.10.0"
+      "version": "2.1.0",
+      "checkout": "v2.1.0"
```
When using `"download"` instead of `"repository"` (see suggestion above), we no longer need to use `"checkout"`:

```suggestion
      "version": "2.1.0"
```

jankeromnes reviewed (approved):
Wow, that's awesome, many thanks @niklaskeerl! 💯 That's super helpful.

Out of curiosity, did you use the `node upgrade-extensions` script?

niklaskeerl commented:
> Out of curiosity, did you use the `node upgrade-extensions` script?

No, I wrote a script to check if each extension has a new tag on GitHub.

```
let Parser = require("rss-parser");
let parser = new Parser();
var extensions = require("./extensions.json");

extensions.extensions.forEach((item) => {
  if (
    item.repository == null ||
    item.version == null ||
    item.checkout == null
  ) {
  } else {
    repo_url = item.repository.replace(".git", "");
    parser
      .parseURL(repo_url + "/tags.atom")
      .then((feed) => {
        if (item.checkout != feed.items[0].link.split("/").pop()) {
          console.log(item.id);
          console.log(item.checkout);
          console.log(feed.items[0].link.split("/").pop());
        }
      })
      .catch((err) => {
        console.log(item.repository + "/tags.atom");
        console.log(err);
      });
  }
});
```
